RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Live.swift

```diff
@@ -0,0 +1,153 @@
+import CloudKit
+import ComposableArchitecture
+import DependenciesAdditions
+import os
+
+extension CKRecord.RecordType {
+	static let profile = "Profile"
+}
+
+extension CKRecord.FieldKey {
+	static let content = "Content"
+}
+
+extension CloudBackupClient {
+	struct IncorrectRecordTypeError: Error {}
+	struct NoProfileInRecordError: Error {}
+	struct ProfileMissingFromKeychainError: Error { let id: UUID }
+
+	public static let liveValue: Self = .live()
+
+//	private static let container = CKContainer(identifier: "iCloud.com.radixpublishing.radixwallet.ios.dev.cloudBackup")
+	private static let container = CKContainer(identifier: "iCloud.com.radixpublishing.radixwallet.ios.pre-alpha.cloudBackup")
+
+	public static func live(
+		profileStore: ProfileStore = .shared
+	) -> CloudBackupClient {
+		@Dependency(\.secureStorageClient) var secureStorageClient
+		@Dependency(\.userDefaults) var userDefaults
+
+		Task {
+			for try await profile in await profileStore.values() {
+				guard profile.appPreferences.security.isCloudProfileSyncEnabled else { continue }
+				let existingRecord = try? await fetchProfileRecord(.init(recordName: profile.id.uuidString))
+				let result: BackupResult.Result
+				do {
+					try await saveProfile(profile, existingRecord: existingRecord)
+					result = .success
+				} catch CKError.accountTemporarilyUnavailable {
+					result = .temporarilyUnavailable
+				} catch CKError.notAuthenticated {
+					result = .notAuthenticated
+				} catch {
+					result = .failure
+				}
+
+				try? userDefaults.setLastCloudBackup(result, of: profile)
+			}
+		}
+
+		@Sendable
+		func fetchProfileRecord(_ id: CKRecord.ID) async throws -> CKRecord {
+			try await container.privateCloudDatabase.record(for: id)
+		}
+
+		@Sendable
+		func fetchAllProfileRecords() async throws -> [CKRecord] {
+			let records = try await container.privateCloudDatabase.records(
+				matching: .init(recordType: .profile, predicate: .init(value: true))
+			)
+			return try records.matchResults.map { try $0.1.get() }
+		}
+
+		@Sendable
+		func extractProfile(_ record: CKRecord) throws -> Profile {
+			guard record.recordType == .profile else {
+				throw IncorrectRecordTypeError()
+			}
+			guard let asset = record[.content] as? CKAsset, let fileURL = asset.fileURL else {
+				throw NoProfileInRecordError()
+			}
+
+			let data = try Data(contentsOf: fileURL)
+			let profile = try Profile(jsonData: data)
+			try FileManager.default.removeItem(at: fileURL)
+
+			return profile
+		}
+
+		@discardableResult
+		@Sendable
+		func saveProfile(_ profile: Profile, existingRecord: CKRecord?) async throws -> CKRecord {
```

Comment:
I think it would be more clear to name it `uploadProfileToCloud` which is what we are doing. At least while reading code in this file, `saveProfile` could have meant `downloadProfileToLocalKeychain`.

Reply:
Yes `save` is a bit confusing, but since the client is called `CloudBackupClient`, it is perhaps enough to name it `backupProfile`.